Rubberduck.Parsing/Symbols/ParameterDeclaration.cs

```diff
@@ -39,43 +41,51 @@ public class ParameterDeclaration : Declaration
             IsByRef = isByRef;
             IsImplicitByRef = false;
             IsParamArray = isParamArray;
+            DefaultValue = defaultValue;
         }
 
         /// <summary>
         /// Creates a new user declaration for a parameter.
         /// </summary>
-        public ParameterDeclaration(QualifiedMemberName qualifiedName, 
+        public ParameterDeclaration(QualifiedMemberName qualifiedName,
             Declaration parentDeclaration,
-            ParserRuleContext context, 
-            Selection selection, 
+            ParserRuleContext context,
+            Selection selection,
             string asTypeName,
             VBAParser.AsTypeClauseContext asTypeContext,
             string typeHint,
             bool isOptional,
             bool isByRef,
-            bool isArray = false, 
+            bool isArray = false,
             bool isParamArray = false,
             bool isUserDefined = true)
             : base(
-                  qualifiedName, 
-                  parentDeclaration, 
+                  qualifiedName,
+                  parentDeclaration,
                   parentDeclaration,
                   asTypeName,
                   typeHint,
-                  false, 
-                  false, 
+                  false,
+                  false,
                   Accessibility.Implicit,
-                  DeclarationType.Parameter, 
-                  context, 
+                  DeclarationType.Parameter,
+                  context,
                   selection,
                   isArray,
                   asTypeContext,
                   isUserDefined)
         {
+            var argContext = context as VBAParser.ArgContext;
             IsOptional = isOptional;
             IsByRef = isByRef;
-            IsImplicitByRef = isByRef && ((VBAParser.ArgContext) context)?.BYREF() == null;
+            IsImplicitByRef = isByRef && argContext?.BYREF() == null;
             IsParamArray = isParamArray;
+
+            if (!isOptional || argContext?.argDefaultValue() == null)
+            {
+                return;
+            }
+            DefaultValue = argContext.GetDescendent<VBAParser.LiteralExprContext>()?.GetText() ?? string.Empty;
```

Comment:
This does not look appropriate. Default members are not restricted to literals, just to compile time constants.

The following is completely legal:
```vba
Private Const bar As Long = 2

Private Enum Foo
   FooBar = bar + 15 - Module1.OtherModuleConst
End Enum


Private Sub DoSomething(Optional baz As Long = bar * 15 + Foo.FooBar + Module1.OtherModuleConst * FooBar)
   Debug.Print baz
End Sub
```
Here, `OtherModuleConst` is a public constant declared in `Module1`.

Until we can evaluate the expression context, I would propose to use `argContext.argDefaultValue().Expression.GetText()`.

Reply:
Not sure why I was thinking it had to be a literal. This is fixed.